crates/sui-indexer-alt/src/bootstrap.rs

```diff
@@ -33,13 +32,22 @@ pub async fn bootstrap(
     indexer: &Indexer<Db>,
     retry_interval: Duration,
     cancel: CancellationToken,
+    stored_genesis: Option<StoredGenesis>,
 ) -> Result<StoredGenesis> {
     info!("Bootstrapping indexer with genesis information");
 
     let Ok(mut conn) = indexer.store().connect().await else {
         bail!("Bootstrap failed to get connection for DB");
     };
 
+    if let Some(stored_genesis) = stored_genesis {
```

Comment:
This should happen after you check whether the database already contains this record.